packages/react-async/README.md

```diff
@@ -0,0 +1,147 @@
+
+Tools for creating powerful, asynchronously-loaded React components.
+
+## Installation
+
+```bash
+$ yarn add @shopify/react-async
+```
+
+## Usage
+
+### `createAsyncComponent()`
+
+`createAsyncComponent` is a function for creating components that are loaded asynchronously on initial mount. However, the resulting component does more than just help you split up your application along component lines; it also supports customized rendering for loading, and creates additional components for smartly preloading or preloading the component’s bundle. Best of all, in conjunction with the Babel and Webpack plugins provided by [`@shopify/async`](https://github.com/Shopify/quilt/tree/master/packages/async), you can easily extract the bundles needed to render your application during server side rendering.
+
+To start, import the `createAsyncComponent` function. The simplest use of this function requires just a `load` function, which returns a promise for a component:
+
+```tsx
+import {createAsyncComponent} from '@shopify/react-async';
+
+const MyComponent = createAsyncComponent({
+  load: () => import('./MyComponent'),
+});
+```
+
+This function returns a component that accepts the same props as the original one.
+
+`createAsyncComponent` also adds a few static members that are themselves components: `Preload`, `Prefetch`, and `KeepFresh`.
+
+```tsx
+const MyComponent = createAsyncComponent({
+  load: () => import('./MyComponent'),
+});
+
+// All of these are available:
+<MyComponent />
+<MyComponent.Prefetch />
+<MyComponent.Preload />
+<MyComponent.KeepFresh />
+```
+
+By default, `Preload`, `Prefetch`, and `KeepFresh` all simply prefetch the bundle for the component in the background. However, you can provide additional markup to render during these phases with the `renderPreload`, `renderPrefetch`, and `renderKeepFresh` options to `createAsyncComponent`:
+
+```tsx
+const MyComponent = createAsyncComponent({
+  load: () => import('./MyComponent'),
+  renderPrefetch: () => <PrefetchSomethingElse />,
+});
+
+// Now prefetches the component, and <PrefetchSomethingElse />!
+<MyComponent.Prefetch />;
+```
+
+While you can supply whatever markup you like for these, we recommend that you use them for the following purposes:
+
+- `Prefetch`: loading resources that will be used by the component
+- `Preload`: loading resources **and** data that will be used by the component
+- `KeepFresh`: loading resources and data that will be used by the component, and keeping data up to date
+
+If you want props for your `Preload`, `Prefetch`, or `KeepFresh` components, simply provide them in the `render` option for that component. The resulting components will have those prop types baked in.
+
+```tsx
+const MyComponent = createAsyncComponent({
+  load: () => import('./MyComponent'),
+  renderPreload: ({priority}: {priority: 'high' | 'low'}) => (
```

Comment:
what does `low` priority mean in this case? Is that more of a `defer` than a `preload`?

Reply:
this is just an example to show that you can include arbitrary props in `renderPreload` and that it will thread those typings through